### ISSUE TITLE
registerBlockType supports multiple - block inserter stuck open

### ISSUE THREAD

**Issue by stuffnting** [WordPress/gutenberg#27586]
**Describe the bug**

With this code, the block inserter does not close after insertion:

```
const { registerBlockType } = wp.blocks;

registerBlockType("snt/broken-app", {
  title: "Broken App",
  icon: "calendar-alt",
  category: "text",
  supports: {
    multiple: false,
  },
  edit: () => <p>Break me!</p>,
  save: () => null,
});
```

**To reproduce**
1. Open the main editor inserter from the toolbar at the top
2. Click on the Broken App block icon to insert the block
3. Block is inserted but the inserter is stuck open
4. Click inserter "+" button, inserter still stuck open
5. Click away from the inserter inside the editor, inserter still stuck open
6. Click on the Broken App icon again, then click away  from the inserter in the editor, inserter now closes.

On 2 console logs:

`[Violation] 'click' handler took 173ms`

On 6 console logs:

`[Violation] 'setTimeout' handler took 71ms`

**Expected behavior**
For all other blocks, and indeed this code if supports `multiple` is changed to `true`, the inserter closes automatically after the block is inserted. 

**Editor version**
- WordPress version: 5.5.3
- no Gutenberg plugin

**Desktop**
Tested on Chrome and Firefox, happens on both.

**Comment by talldan** [comment]
@stuffnting I wasn't able to reproduce this on 5.5.3 or 5.6 (I edited the gif a little short, but you can just about see that the menu closes):
![broken](https://user-images.githubusercontent.com/677833/101569222-00f66400-3a0f-11eb-80b9-b8bab0510934.gif)

Anything I'm missing in my testing? Would you be able to share a screenshot of the inserter when it's stuck open?

**Comment by stuffnting** [comment]
Thank you for the prompt response. Having seen your screenshot, I went back and retested. 

It only happens with the top left inserter button, but not with the button that you were using in the block area of the editor. 

Note, that on the screenshot below, the inserter only closes again after I have clicked the Broken App icon  a second time.

I have tried it with no plugins, and get the same behaviour.

One other thing, I forgot to say it is a network site, but I am not sure if that would make any difference.

![broken-app](https://user-images.githubusercontent.com/47080454/101682545-aa634700-3a5b-11eb-9259-6c7cc7f5b81c.gif)

**Comment by talldan** [comment]
Cheers for the extra info, I can reproduce that now so I've labelled as a bug.

It'll be something to do with the way focus moving outside the menu is detected, the block icon becoming disabled must cause an issue.

A workaround is to click the 'Search' input at the top and then click back outside of the inserter (moving focus back inside makes the focus detection work properly again.)

**Comment by stuffnting** [comment]
Thank you talldan.

I am glad to know I have not completely lost the plot.
